cpp/CMakeLists.txt

```diff
@@ -532,9 +539,11 @@ set(ARROW_STATIC_LINK_LIBS
   brotli_enc
   brotli_common
   snappy
-  zlib
-  zstd_static
-  lz4_static)
+  zlib)
+
+if(NOT ARROW_BUILD_PARQUET_DEPS_ONLY)
+  list(APPEND ARROW_STATIC_LINK_LIBS zstd_static lz4_static)
+endif()
```

Comment:
Can you add variables to toggle adding these libraries to the build? If ARROW_BUILD_PARQUET_DEPS_ONLY then they should be toggled off